bundles/extensions/i18n/src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -197,9 +197,19 @@ public void handleEvent(final org.osgi.service.event.Event event) {
             // invalidate everything
             if (languageRootPaths.contains(path)) {
                 log.debug(
-                        "handleEvent: Detected change of cached language root '{}', removing all cached ResourceBundles",
+                        "handleEvent: Detected change of cached language root '{}', removing all cached ResourceBundle that are affected",
                         path);
-                scheduleReloadBundles(true);
+
+                for (JcrResourceBundle bundle : resourceBundleCache.values()) {
```

Comment:
@kwin IMO that will not be as frequent case as updating a translation So in that case we can discard  the complete cache. Will that be okay ?